File: Data Structures & Algorithms/valid-parenthesis-string/submission-9.cs
public class Solution {
    public bool CheckValidString(string s) {
        return Dfs(0, 0, s);
    }

    private bool Dfs(int i, int open, string s) {
        if (open < 0) return false;
        if (i == s.Length) return open == 0;

        if (s[i] == '(') {
            return Dfs(i + 1, open + 1, s);
        } else if (s[i] == ')') {
            return Dfs(i + 1, open - 1, s);
        } else {
            return Dfs(i + 1, open, s) ||
                   Dfs(i + 1, open + 1, s) ||
                   Dfs(i + 1, open - 1, s);
        }
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-9.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens) {
        const dfs = () => {
            const token = tokens.pop();
            if (!"+-*/".includes(token)) {
                return parseInt(token);
            }

            const right = dfs();
            const left = dfs();

            if (token === '+') {
                return left + right;
            } else if (token === '-') {
                return left - right;
            } else if (token === '*') {
                return left * right;
            } else {
                return Math.trunc(left / right);
            }
        };

        return dfs();
    }
}
